src/odemis/acq/fastem.py

```diff
@@ -642,16 +646,24 @@ def acquire_roa(self, dataflow):
 
             self.move_stage_to_next_tile()  # move stage to next field image position
             self._scanner.blanker.value = False  # unblank the beam
-            # TODO remove the below temporary fix when a proper solution is found
-            # The temporary fix has been applied so that the ROAs acquisition can still continue upon failure
-            # of correcting the beam shift
-            try:
-                self.correct_beam_shift()  # correct the shift of the beams caused by the parasitic magnetic field.
-            except Exception:
-                logging.exception("Correcting the beam shift failed, check if the image quality is still good.")
-                # In case of failure save the ccd image
-                ccd_image = self._ccd.data.get(asap=False)
-                fastem_util.save_image(self.path, f"{self.field_idx}_after.tiff", ccd_image)
+
+            prev_beam_shift = self._beamshift.shift.value
+            if field_idx in beam_shift_indices or beam_shift_failed:
+                logging.debug(f"Will run beam shift correction for field index {field_idx}")
+                try:
+                    new_beam_shift = self.correct_beam_shift()
+                    # The difference in x or y should not be larger than 2 micrometers
+                    if any(map(lambda x, y: abs(x - y) > 2e-6, new_beam_shift, prev_beam_shift)):
```

Comment:
The names of the variables are confusing. It feels you're comparing X with Y. x & y refer to the new and previous values. This is computed for all axes, both X and Y axes. Right?
```suggestion
                    if any(map(lambda a, b: abs(a - b) > 2e-6, new_beam_shift, prev_beam_shift)):
```
(or maybe name them "n" and "p"?)

Reply:
Or maybe just compare the complete distance, with something like:
```suggestion
                    if math.hypot(new_beam_shift[0] prev_beam_shift[0], new_beam_shift[1] - prev_beam_shift[1]) >  2e-6:
```

Reply:
The names are indeed confusing, I'll leave it as a per axis check though, during testing we always see that the distance in one direction is high, and in the other not. So I think it is a safe check.